canonical-web-and-design/maas-ui
# refactor(ui): script slice

## Done

- Migrate the actions and reducers for scripts to a slice.
- Replaced HTTP API methods with the WS equivalents.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to Settings -> Commissioning scripts (or Testing scripts).
- You should see a list of scripts.
- Open the Upload form and upload a script. Save this as a script if you need one:
```
#!/usr/bin/env python3

# --- Start MAAS 1.0 script metadata ---
# name: example
# parallel: instance
# parameters:
#   storage: {type: storage}
# --- End MAAS 1.0 script metadata ---

import argparse

parser = argparse.ArgumentParser(description='')
parser.add_argument(
    '--storage', dest='storage', required=True,
    help='path to storage device you want to test. e.g. /dev/sda')
args = parser.parse_args()

print("Testing: %s" % args.storage)

```
- You should get returned to the list of scripts and the new script should appear.
- Delete the script and it should get removed from the list.

## Fixes

Fixes: #1408.
